Microsoft.HealthVault.Fhir/Transformers/ObservationToHealthVault.cs

```diff
@@ -66,24 +70,52 @@ public static ThingBase ToHealthVault(this Observation observation)
             return baseThing;
         }
 
-        internal static T GetThingValueFromQuantity<T>(Quantity value) where T : Measurement<double>, new()
+        internal static T GetThingValueFromQuantity<T>(Quantity quantityValue) where T : Measurement<double>, new()
         {
-            // TODO: detect the units from the code (value.Unit)
-            if (value != null)
+            if (quantityValue?.Value == null)
             {
-                if (value.Value.HasValue)
+                return null;
+            }
+            
+            var unitConversion = UnitResolver.Instance.UnitConversions.FirstOrDefault(x => x.Code.Equals(quantityValue.Code, StringComparison.Ordinal));
+
+            double convertedValue;
+            if (unitConversion != null)
+            {
+                object unitEnum;
+                try
```

Comment:
The goal of this is to check that the UnitsNetUnitEnum exists in the UnitsNet.Units namespace and that the UnitsNetSource is able to be parsed out of that namespace. The errors that get thrown if you mistype are the same as if you didn't supply the parameter.